kooplex-proxy/proxy/models.py

```diff
@@ -1,31 +1,48 @@
-from sqlalchemy import Boolean, Column, ForeignKey, Integer, String, Table, MetaData
+from sqlalchemy import Boolean, Column, ForeignKey, Integer, String, Table
 from sqlalchemy.dialects.postgresql import DOUBLE_PRECISION, TEXT, TIMESTAMP, \
     VARCHAR, INTEGER, REAL, BOOLEAN, DATE, NUMERIC
-from database import Base
+from sqlalchemy.ext.declarative import declarative_base
 
-METADATA = MetaData()
+from database import ALLOWED_SCHEMAS
 
+Base = declarative_base()
 
-class MViewCountrySamples(Base):
+
+class AbstractBase(Base):
+
+    __abstract__ = True
+    @classmethod
+    def get_schema(cls):
+        return cls.__table_args__['schema']
+
+    @classmethod
+    def set_schema(cls, schema_name):
+        cls.__table_args__['schema'] = schema_name
+
+
+class MViewCountrySamples(AbstractBase):
     __tablename__ = "app_country_samples_full"
+    __table_args__ = {'schema': ALLOWED_SCHEMAS['schema_1']}
```

Comment:
I was thinking that this line "__table_args__ = {'schema': ALLOWED_SCHEMAS['schema_1']}" could be in the parent class so it gets inherited by child classes.

Reply:
yes, indeed